### PR TITLE
Issue 310 : Enable inflight event configuration on Python writer API.

### LINKED ISSUE #310: Performance improvements for Python Event Writer API
**Problem description**
At the moment Python API to write events into Pravega ensures only one inflight event at any given point in time. This causes a drop in the performance of the Python API.

**Problem location**
Python Event Writer

**Suggestions for an improvement**
Ensure we can have multiple inflight events with the python writer APIs.

### PR DESCRIPTION
**Pull request by shrids**
**Change log description**  
- Enable a configurable number of inflight events to be written by the Python writer.

**Purpose of the change**  
Fixes #310 

**What the code does**  
- Create a oneshot receiver holder which tracks multiple oneshot receivers tracking every inflight event.
- Users will now be able to specify the max number of inflight events that can be configured for the Python writer.
```rust
  pub fn create_writer(
        &self,
        scope_name: &str,
        stream_name: &str,
        max_inflight_events: usize,
    ) -> PyResult<StreamWriter> {..}
```
- A flush () API has been enabled for the Python Writer which ensures all the inflight events are pesisted.

``` rust
///
/// Flush all the inflight events into Pravega Stream.
/// This will ensure all the inflight events are completely persisted on the Pravega Stream.
///
pub fn flush(&mut self) {...}
```

**How to verify it**  
All the existing and newly added tests should pass.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -55,6 +55,6 @@ pub(crate) mod segment;
 #[doc(hidden)]
 pub mod test_utils;
 #[macro_use]
-mod util;
+pub mod util;
```
Comment (Tristan1900) [inline src/lib.rs:58]: We probably want to hide this module in document. Like `#[doc(hidden)]`

File: python_binding/src/stream_writer.rs [hunk 2]
```diff
@@ -137,12 +157,31 @@ impl StreamWriter {
...
+    ///
+    /// Flush all the inflight events into Pravega Stream.
+    /// This will ensure all the inflight events are completely persisted on the Pravega Stream.
+    ///
+    #[text_signature = "($self)"]
+    #[args("*")]
+    pub fn flush(&mut self) {
+        info!("Invoking flush() on writer {:?}", self.to_str());
+        for x in self.inflight.drain() {
+            let _res = self.factory.runtime().block_on(x);
```
Comment (Tristan1900) [inline python_binding/src/stream_writer.rs:169]: We are ignoring the oneshot result. What if the result is `Err`, I think we should bubble it up to the caller or throw an exception.

File: src/lib.rs (hunk 1 above)
Comment (shrids) [reply src/lib.rs:58]: done.

File: python_binding/src/stream_writer.rs (hunk 2 above)
Comment (shrids) [reply python_binding/src/stream_writer.rs]: yes, I have improved the code.

File: src/util/oneshot_holder.rs [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+//
+
+use std::collections::vec_deque::Drain;
+use std::collections::VecDeque;
+use tokio::sync::oneshot::error::RecvError;
+use tokio::sync::oneshot::Receiver;
+
+pub struct OneShotHolder<E> {
+    size: usize,
+    inflight: VecDeque<tokio::sync::oneshot::Receiver<Result<(), E>>>,
```
Comment (tkaitchuck) [inline src/util/oneshot_holder.rs:18]: Receiver is imported. No need to specify the full path.

**Review by tkaitchuck — APPROVED** [review]

File: src/util/oneshot_holder.rs (hunk 3 above)
Comment (shrids) [reply src/util/oneshot_holder.rs]: removed the full path.

**Review by Tristan1900 — APPROVED** [review]

**Pull request merged**